base_layer/core/src/transactions/transaction_protocol/sender.rs

```diff
@@ -482,6 +485,11 @@ pub fn calculate_tx_id<D: Digest>(pub_nonce: &PublicKey, index: usize) -> u64 {
     u64::from_le_bytes(bytes)
 }
 
+pub(crate) fn calculate_total_amount(amount_to_self: MicroTari, amounts: Vec<MicroTari>) -> MicroTari {
```

Comment:
Assuming this function lives on (and moves), you don't want to take ownership of `amount` here, otherwise there's no much unnecessary memory copying. Take a reference, either `&[u8]` or `&FixedSet<MicroTari>`

Reply:
Noted, thanks. Reverting to the former.